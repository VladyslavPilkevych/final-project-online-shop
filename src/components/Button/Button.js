import React, { memo } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import PropTypes from 'prop-types';
import styles from './Button.module.scss';

function Button(props) {
  const {
    children, onClick, type, style, className,
  } = props;

  const user = useSelector((state) => state.user.user);
  const disable = Boolean(user);

  return (
    /* eslint-disable react/button-has-type */
    <button
      type={type}
      onClick={onClick}
      className={`${className} ${style}`}
      // disable={!disable}
    >
      {children}
    </button>
    /* eslint-enable react/button-has-type */
  );
}

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  type: PropTypes.oneOf(['submit', 'button']),
  onClick: PropTypes.func,
  style: PropTypes.string,
  className: PropTypes.string,
};

Button.defaultProps = {
  children: '',
  type: 'button',
  onClick: () => {},
  style: '',
  className: '',
};

export default memo(Button);
