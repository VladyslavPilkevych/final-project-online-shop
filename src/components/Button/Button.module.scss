@import '../../assets/styles/global.module.scss';

.editBtn {
  font-weight: 600;
  font-size: 14px;
  line-height: 21px;
  border: 2px solid $blue;
  box-sizing: border-box;
  border-radius: 50px;
}
