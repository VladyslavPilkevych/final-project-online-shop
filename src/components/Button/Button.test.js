import { render } from '@testing-library/react';
import Button from './Button';

describe('Snapshot Button should render', () => {
    test('Snapshot Button is rendered', () => {
      const { asFragment } = render(<Button />);
      expect(asFragment()).toMatchSnapshot();
    });
  });