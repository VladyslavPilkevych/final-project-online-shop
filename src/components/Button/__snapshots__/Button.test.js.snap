// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Button should render Snapshot Button is rendered 1`] = `
<DocumentFragment>
  <button
    class=" [object Object]"
    type="button"
  />
</DocumentFragment>
`;
